packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -101,6 +101,125 @@ ChartInterval chartInterval(String displayName) {
   }
 }
 
+class AllocationStackTrace {
+  AllocationStackTrace(CpuSample sample, {List<ProfileFunction> functions}) {
+    _computeStacktrace(sample, functions: functions);
+  }
+
+  int timestamp;
+
+  var showFullStacktrace = false;
+
+  final stacktrace = <String>[];
+
+  final sources = <String>[];
+
+  int get stackDepth => stacktrace.length;
+
+  static const NoClasses = 0;
+  static const DartClasses = 1;
+  static const FlutterClasses = 2;
+
+  static const DartFlutterClasses = DartClasses | FlutterClasses;
+
+  bool hideDartClasses(int hideClasses) =>
```

Comment:
Fyi @kenzieschmoll I think you have wanted to filter stack traces from the Dart core libraries and Flutter as well. @terrylucas is this method used currently? I suspect implementing it will be hard as the stacktraces currently lack the info needed.
One other item to hide and show is native frames.
This is something we should filter consistently across DevTools.

Reply:
This is currently controlled by the filter dialog in memory profiler.  Should this filtering be a a higher level (top-level setting)?  Then all stack traces, classes displayed in snapshots, allocation monitoring, etc. would use the same filtering.

I'll add an issue https://github.com/flutter/devtools/issues/2685